routes/reportRoutes.ts

```diff
@@ -0,0 +1,280 @@
+import { Request, Response } from "express";
+import { asc, desc, eq, isNotNull, sql } from "drizzle-orm";
+import {
+  categoryTable,
+  db,
+  materialCategoriesTable,
+  materialRecommendationsTable,
+  materialTable,
+  menteeTable,
+} from "../drizzle";
+
+// Reports endpoints
+
+// Get all mentees with related materials
+async function getAllMenteesWithMaterials(
+  req: Request,
+  res: Response
+): Promise<void> {
+  try {
+    const result = await db
+      .select({
+        menteeId: menteeTable.id,
+        firstName: menteeTable.firstName,
+        lastName: menteeTable.lastName,
+        url: materialTable.url,
+        type: materialTable.type,
+      })
+      .from(menteeTable)
+      .innerJoin(materialTable, eq(menteeTable.id, materialTable.menteeId));
+
+    res.send(result);
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Error in get all mentees with materials");
+  }
+}
+
+// Get all materials with related categories
+async function getAllMaterialsWithCategories(
+  req: Request,
+  res: Response
+): Promise<void> {
+  try {
+    const result = await db
+      .select({
+        materialUrl: materialTable.url,
+        materialType: materialTable.type,
+        categoryName: categoryTable.name,
+        categoryDescription: categoryTable.description,
+      })
+      .from(materialTable)
+      .leftJoin(
+        materialCategoriesTable,
+        eq(materialTable.id, materialCategoriesTable.materialId)
+      )
+      .leftJoin(
+        categoryTable,
+        eq(categoryTable.id, materialCategoriesTable.categoryId)
+      );
+
+    res.send(result);
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Error in get all materials with categories");
+  }
+}
+
+// Get all categories with related materials
+async function getAllCategoriesWithMaterials(
+  req: Request,
+  res: Response
+): Promise<void> {
+  try {
+    const result = await db
+      .select({
+        categoryName: categoryTable.name,
+        categoryDescription: categoryTable.description,
+        materialUrl: materialTable.url,
+        materialType: materialTable.type,
+      })
+      .from(materialTable)
+      .rightJoin(
+        materialCategoriesTable,
+        eq(materialTable.id, materialCategoriesTable.materialId)
+      )
+      .rightJoin(
+        categoryTable,
+        eq(categoryTable.id, materialCategoriesTable.categoryId)
+      );
+
+    res.send(result);
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Error in get all categories with materials");
+  }
+}
+
+// Get all materials with categories or recommendations (distinct)
+async function getAllUsedMaterialsDistinct(
+  req: Request,
+  res: Response
+): Promise<void> {
+  try {
+    const result = await db
```

Comment:
Could this be done with a union as documented here? https://orm.drizzle.team/docs/set-operations

You might be able to start by querying materialCategories and materialRecommendations:

```
const materialCategoriesQuery = db.select({ materialId: materialTable.id }).from(materialCategoriesTable)
const materialRecommendationsQuery = db.select({ materialId: materialTable.id }).from(materialRecommendationsTable)
const result = await union(materialCategoriesQuery, mateiralRecommendationsQuery)
```

Then use that result to get the right list of materials. The user story doesn't actually request you to list the categories and recommendations.